### PR TITLE
Fixes for MSVC

### PR DESCRIPTION
**Pull request by ptoharia**

### CODE REVIEW

**Comment by ptoharia** [comment]
Please don't merge yet, I have some further comments

**Comment by ptoharia** [comment]
Still having some errors/warnings not sure how to proceed:

- brain\spikeReportReader.cpp(92): warning C4056: overflow in floating point constant arithmetic --> I think this warning could be disabled, but disabling it for the file or some CMake/common global configuration for MSVC.

- compartmentConverter.cpp(187): error C3016:  index variable in OpenMP 'for' statement must have signed integral type --> could be easily be fixed changing size_t for int, it's this ok?

- Linking error (Boost 1.56)  Any ideas on this? Might be Boost version?
"class std::basic_string<char,struct std::char_traits<char>,class std::allocator<char> > boost::program_options::arg" (?arg@program_options@boost@@3V?$basic_string@DU?$char_traits@D@std@@V?$allocator@D@2@@std@@A) not resolved."   

- HighFive has also a minor warning in the tag Brion is using. Tried to fix it on HighFive's HEAD but got quite a few more warnings/errors.

File: brain/circuit.cpp [hunk 1]
```diff
@@ -109,7 +115,7 @@ class Circuit::Impl
         brain::GIDSet gids;
         brain::GIDSet::const_iterator hint = gids.begin();
         for( size_t i = 0; i < getNumNeurons(); ++i )
-            hint = gids.insert( hint, i + 1 );
+          hint = gids.insert( hint, (( unsigned int ) i ) + 1 );
```
Comment (tribal-tec) [inline brain/circuit.cpp]: please use uint32_t( i ) instead, also for the other places

**Comment by tribal-tec** [comment]
1. Not sure about the overflow, could be an actual issue, maybe @hernando can help?
2. Yes, int is fine then
3. Where does the error come from? library? executable? I haven't seen this one yet.
4. We have warnings as well with HighFive on Linux, but we ignore the silently :)

File: brain/circuit.cpp (hunk 1 above)
Comment (ptoharia) [inline brain/circuit.cpp]: done

**Comment by ptoharia** [comment]
@tribal-tec 

1. Might not be safe to ignore it, read in the doc (https://msdn.microsoft.com/en-us/library/cwt7tyxx.aspx) that it should be safe if it dissapears disabling optimizations, which I did and still got the warning

2. Done

3. Linking executables. For example (sorry some of it is in Spanish):
 C:\Program Files (x86)\Microsoft Visual Studio 12.0\VC\bin\x86_amd64\link.exe /ERRORREPORT:QUEUE /OUT:"C:\Users\ptoharia\Brio n\b\bin\Debug\morphologyConverter.exe" /INCREMENTAL /NOLOGO /LIBPATH:"C:/libs/boost_1_56_0/lib64-msvc-12.0" /LIBPATH:"C:/libs /boost_1_56_0/lib64-msvc-12.0/Debug" kernel32.lib user32.lib gdi32.lib winspool.lib shell32.lib ole32.lib oleaut32.lib uuid.l ib comdlg32.lib advapi32.lib ..\lib\Debug\Brion_D.lib "C:\libs\boost_1_56_0\lib64-msvc-12.0\boost_program_options-vc120-mt-gd -1_56.lib" ..\lib\Debug\Lunchbox_D.lib "C:\libs\boost_1_56_0\lib64-msvc-12.0\boost_filesystem-vc120-mt-gd-1_56.lib" "C:\libs\ boost_1_56_0\lib64-msvc-12.0\boost_regex-vc120-mt-gd-1_56.lib" "C:\libs\boost_1_56_0\lib64-msvc-12.0\boost_serialization-vc12 0-mt-gd-1_56.lib" "C:\libs\boost_1_56_0\lib64-msvc-12.0\boost_system-vc120-mt-gd-1_56.lib" "C:\libs\boost_1_56_0\lib64-msvc-1 2.0\boost_thread-vc120-mt-gd-1_56.lib" "C:\libs\boost_1_56_0\lib64-msvc-12.0\boost_unit_test_framework-vc120-mt-gd-1_56.lib" "C:\libs\boost_1_56_0\lib64-msvc-12.0\boost_chrono-vc120-mt-gd-1_56.lib" "C:\libs\boost_1_56_0\lib64-msvc-12.0\boost_date_tim e-vc120-mt-gd-1_56.lib" "C:\libs\boost_1_56_0\lib64-msvc-12.0\boost_atomic-vc120-mt-gd-1_56.lib" ..\lib\Debug\Servus_D.lib /M ANIFEST /MANIFESTUAC:"level='asInvoker' uiAccess='false'" /manifest:embed /DEBUG /PDB:"C:/Users/ptoharia/Brion/b/bin/Debug/mo rphologyConverter.pdb" /SUBSYSTEM:CONSOLE /TLBID:1 /DYNAMICBASE /NXCOMPAT /IMPLIB:"C:/Users/ptoharia/Brion/b/lib/Debug/morphologyConverter.lib" /MACHINE:X64  /machine:x64 morphologyConverter.dir\Debug\morphologyConverter.obj
morphologyConverter.obj : error LNK2001: sÝmbolo externo "class std::basic_string<char,struct std::char_traits<char>,class std: :allocator<char> > boost::program_options::arg" (?arg@program_options@boost@@3V?$basic_string@DU?$char_traits@D@std@@V?$allocat or@D@2@@std@@A) sin resolver [C:\Users\ptoharia\Brion\b\apps\morphologyConverter.vcxproj] morphologyConverter.obj : error LNK2001: sÝmbolo externo "public: static unsigned int const boost::program_options::options_des cription::m_default_line_length" (?m_default_line_length@options_description@program_options@boost@@2IB) sin resolver [C:\Users \ptoharia\Brion\b\apps\morphologyConverter.vcxproj] C:\Users\ptoharia\Brion\b\bin\Debug\morphologyConverter.exe : fatal error LNK1120: 2 externos sin resolver [C:\Users\ptoharia\B rion\b\apps\morphologyConverter.vcxproj]

4. Sorry, but don't know how to solve this cause I don't see a find_package to add "system", for example. Can you give me a hint?

5. (new one) /Wextra not valid for MVDTool. I'm aware this is solved in CMake/common but MSVDTool is using MVDTool/CMake/config/ReleaseDebugAutoFlags.cmake for compiler options.

Thanks!

**Comment by tribal-tec** [comment]
3. It misses the famous BOOST_PROGRAM_OPTIONS_DYN_LINK define I guess.  Try `target_compile_definitions(morphologyConverter BOOST_PROGRAM_OPTIONS_DYN_LINK)`
4. True. It seems it just works because of the subproject build, hence it knows the target MVDTool like this. Is there a chance you can fix all the warnings and submit a PR for MVDTool?

**Comment by ptoharia** [comment]
@tribal-tec 

1. nice! it worked!
2. I'll try 

Everything fixed except the overflow warning and MVDTool/Highfive stuff

**Comment by hernando** [comment]
The overflow warning is irrelevant because the underflow can only occur is we have a report whose first spikes are the minimum negative float possible, but I think it's better to fix it rather than silencing it. Try adding something like:

    if( time != brion::UNDEFINED_TIMESTAMP && time > -std::numeric_limits< float >::max( ))

**Comment by ptoharia** [comment]
@hernando thanks, tried it but the warning is still there

**Comment by tribal-tec** [comment]
Fine with me

**Comment by ptoharia** [comment]
@tribal-tec done. Just MVDTool/HighFive fixes. When its ready, whats the recomended procedure for updating submodules version?

**Comment by tribal-tec** [comment]
Sorry for the late reply.

Just update the MVDTool SHA-1 in the .gitsubprojects here, that's all.

**Comment by hernando** [comment]
What's the status of this PR?

**Comment by ptoharia** [comment]
@hernando I haven't had time to advance in this and not sure when I'll be able to

**Comment by tribal-tec** [comment]
Please reopen if needed

**Pull request closed without merging**